pkg/controller/atlasdatafederation/datafederation_controller.go

```diff
@@ -170,6 +151,34 @@ func (r *AtlasDataFederationReconciler) Reconcile(context context.Context, req c
 	return workflow.OK().ReconcileResult(), nil
 }
 
+func (r *AtlasDataFederationReconciler) handleDelete(ctx *workflow.Context, log *zap.SugaredLogger, dataFederation *akov2.AtlasDataFederation, project *akov2.AtlasProject, atlasClient *mongodbatlas.Client) workflow.Result {
+	if customresource.HaveFinalizer(dataFederation, customresource.FinalizerLabel) {
+		if customresource.IsResourcePolicyKeepOrDefault(dataFederation, r.ObjectDeletionProtection) {
+			log.Info("Not removing AtlasDataFederation from Atlas as per configuration")
+		} else {
+			if err := r.deleteConnectionSecrets(ctx.Context, dataFederation); err != nil {
+				log.Errorf("failed to remove DataFederation connection secrets from Atlas: %s", err)
```

Comment:
This is inncurate, this is removing from Kubernetes, not Altas yet. I suggest to be a bit imprecise at this point:
```suggestion
				log.Errorf("failed to remove DataFederation connection secrets: %s", err)
```